api-client-generator/templates/typescript-fetch/runtime.mustache

```diff
@@ -115,11 +111,12 @@ export const COLLECTION_FORMATS = {
     pipes: "|",
 };
 
-export type FetchAPI = {{#typescriptThreePlus}}WindowOrWorkerGlobalScope{{/typescriptThreePlus}}{{^typescriptThreePlus}}GlobalFetch{{/typescriptThreePlus}}['fetch'];
+
+export type FetchAPI = (url: string, init?: RequestInit) => Promise<Response>
```

Comment:
> Hmmm I'm doing some more thinking on this and I think there's two ways to go about this:
> 
> 1. Couple the generated API client with the SDK
>    In this case, we could take the generator and move it into the SDK because that's the only place that the generated things are useful, since they require the custom fetch logic. This might make the most sense, as I don't think we're planning on shipping the generated types separately.
> 2. Keep the generated API client agnostic of the caller
>    This requires keeping the generated API client capable of being used with raw fetch, hardcoded base URL (DN), and no retry logic. I'd argue it also should allow the consumer to look at the full body of the response if they want, not just the data field.
> 
> I think right now if we merge this PR we're maybe somewhere in-between: The generated API client requires enough customization of the fetchAPI that maybe keeping it agnostic isn't very valuable (maybe it wasn't valuable to begin with though).
> 
> I gotta run but those are my loose thoughts for now... will come back and revisit tomorrow

I started leaning towards #1 because, for us to use the existing `_makeRequest`, these changes are necessary unless we marshal the `_makeRequest` result into an object with a status on it. I can definitely try that approach but it seems kinda hacky. 

I was kinda working under the assumption that the sdk would be the only consumer of the generated api/types, but I guess potentially there could be other consumers. Let me see if I can get a clean agnostic version, otherwise I would be down to move the api generator into the sdk as well

Reply:
It was using `GlobalFetch` (not sure why actually because we are using typescript v4) so I initially added a declaration for GlobalFetch and matched the existing contract, so yes it's very doable. But after making changes to `request` anyway I thought it would be cleaner to define the custom type inline. Ideally needs to have a type parameter to define the return type

If we do decide to go the agnostic route I'll revert this